src/components/sections/CountryCard/CountryCard.tsx

```diff
@@ -1,25 +1,47 @@
+import { useData } from "../../../data/DataContext";
+import { Link } from "react-router-dom";
+
 const CountryCard = () => {
+
+    const { countries, filtredRegion, typedCountryName } = useData();
+
+    const filteredCountries = countries.filter(country => {
+ 
+        const regionFilter = !filtredRegion || country.region === filtredRegion;
+
+        const nameFilter = !typedCountryName || country.name.common.toLowerCase().startsWith(typedCountryName.toLowerCase());
+
+        return regionFilter && nameFilter;
+    });
+
+
     return (
-        <li className="country-card">
-            <div className="flag"></div>
-            <div className="country-data">
-                <div className="name">
-                    <span>Germany</span>
-                </div>
-                <div className="population">
-                    <span className="population-label label">Population: </span>
-                    <span className="population-data data">81,770,900</span>
-                </div>
-                <div className="region">
-                    <span className="region-label label">Region: </span>
-                    <span className="region-data data">Europe</span>
-                </div>
-                <div className="capital">
-                    <span className="capital-label label">Capital: </span>
-                    <span className="capital-data data">Berlin</span>
-                </div>
-            </div>
-        </li>
+        <>
+            {filteredCountries.map((country, index) => (
```

Comment:
This component is called CountryCard and you render the lint here? I don't understand :D

Reply:
Could please explain what do you mint by "render the lint"?